software/src/enso/pipe.cpp

```diff
@@ -250,16 +252,16 @@ RxTxPipe* Device::AllocateRxTxPipe(bool fallback) noexcept {
   return pipe;
 }
 
-// TODO(sadok): DRY this code.
-RxPipe* Device::NextRxPipeToRecv() {
+struct RxNotification* Device::NextRxNotif() {
```

Comment:
Modified this to let Shinkansen get the actual notification instead of just the pipe